### PR TITLE
Controls-Tech-DBW Integration - `2-15-test-day`

### PR DESCRIPTION
**Pull request by giuseppequaratino**
Merge in encoder_odom and techbus ros nodes into dev. Also remove some unused ros packages in an effort to start cleaning the tech stack

### CODE REVIEW

File: can/can.yml
Comment (giuseppequaratino) [inline can/can.yml]: outdated comment

File: can/can.yml
Comment (giuseppequaratino) [inline can/can.yml]: unused message meant for testing

File: ros/docker/encoder_odom/src/encoder_arduino.py
Comment (giuseppequaratino) [inline ros/docker/encoder_odom/src/encoder_arduino.py:1]: This file hasn't been used for testing yet. Needs to incorporate the changes from encoder_odom.py as well so probably shouldn't merge it.

File: .editorconfig
Comment (dmezh) [inline .editorconfig:1]: Why deleted?

File: Makefile
Comment (dmezh) [inline Makefile]: - [ ] Change version to non-hack before merge

File: ctrl/velocity-controller/ctrl.py
Comment (dmezh) [inline ctrl/velocity-controller/ctrl.py]: Should these also start off as floats?

File: ctrl/velocity-controller/vel.py
Comment (dmezh) [inline ctrl/velocity-controller/vel.py:13]: imo these names could be more clear; it's not apparent what `X_TO_SLOPE_MAPPING` means

File: dbw/py-steering/base.py
Comment (dmezh) [inline dbw/py-steering/base.py:85]: Is this staying or going?

File: dbw/py-steering/steer.py
Comment (dmezh) [inline dbw/py-steering/steer.py:134]: Change this back once we get signed support merged?

File: ros/docker-compose.yml
Comment (dmezh) [inline ros/docker-compose.yml:33]: :)

File: ros/docker-compose.yml
Comment (dmezh) [inline ros/docker-compose.yml]:
```suggestion
        build: docker/techbus/
```

File: ros/docker/encoder_odom/Dockerfile
Comment (dmezh) [inline ros/docker/encoder_odom/Dockerfile:13]: Docker caches in between these RUN directives, meaning the `apt-get update` line could be cached and then the `apt-get install` line runs on a cached version of the apt lists. You need to run update in the same RUN directive:

```suggestion
RUN apt-get update && apt-get install -yq python3-tf2-ros python3-tf-conversions
```

File: ros/docker/encoder_odom/src/encoder_odom.py
Comment (dmezh) [inline ros/docker/encoder_odom/src/encoder_odom.py:13]: order imports

File: ros/docker/encoder_odom/src/encoder_odom.py
Comment (dmezh) [inline ros/docker/encoder_odom/src/encoder_odom.py:52]: yes, you should wait to publish encoder data until you see the first tick and set the initial values to it.

File: ros/docker/encoder_odom/src/encoder_odom.py
Comment (dmezh) [inline ros/docker/encoder_odom/src/encoder_odom.py]:
```suggestion
        tick_distance = circumference/enc_res
```

File: ros/docker/encoder_odom/src/encoder_odom.py
Comment (dmezh) [inline ros/docker/encoder_odom/src/encoder_odom.py]: remove?

File: ros/docker/pure_pursuit/src/global_ros_pp.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/global_ros_pp.py:106]: An exception is thrown and then immediately caught, so is there a point? In other words, what do you actually want to do when we have no intersection?

File: ros/docker/pure_pursuit/src/global_ros_pp.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/global_ros_pp.py:116]: Please make spacing more consistent

File: ros/docker/pure_pursuit/src/global_ros_pp.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/global_ros_pp.py:192]: Missing newline at end of file

File: ros/docker/pure_pursuit/src/path_gen.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/path_gen.py]: Missing newline at end of file

File: ros/docker/pure_pursuit/src/ppwork.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/ppwork.py]: Which one of the pure pursuit controller files is correct?

File: ros/docker/pure_pursuit/src/path_gen.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/path_gen.py]: Maybe a different name for this file would be good - `path_test_gen.py`?

File: ros/docker/pure_pursuit/src/ros_pp.py
Comment (dmezh) [inline ros/docker/pure_pursuit/src/ros_pp.py:1]: A third pure pursuit controller file - which one is right?

File: ros/docker/techbus/Dockerfile
Comment (dmezh) [inline ros/docker/techbus/Dockerfile:24]:
```suggestion
```

File: ros/docker/techbus/Dockerfile
Comment (dmezh) [inline ros/docker/techbus/Dockerfile:17]:
```suggestion

# install redis
```

File: ros/docker/techbus/src/techbus.py
Comment (dmezh) [inline ros/docker/techbus/src/techbus.py:33]:
```suggestion
            rospy.loginfo(f"Warning: invalid steering angle/linear velocity combo: lin.x:{msg.linear.x}, ang.z:{msg.angular.z}")
```

File: ros/docker/techbus/src/techbus.py
Comment (dmezh) [inline ros/docker/techbus/src/techbus.py]: We should make the message `DBW_steeringAngleCmd`

File: ros/docker/techbus/src/techbus.py
Comment (dmezh) [inline ros/docker/techbus/src/techbus.py]: Missing newline at end of file

File: ros/test-scripts/calibrate-start.sh
Comment (dmezh) [inline ros/test-scripts/calibrate-start.sh:83]: Missing newline at end of file

File: ros/test-scripts/cand-start.sh
Comment (dmezh) [inline ros/test-scripts/cand-start.sh:24]: Missing newline at end of file

File: startup.sh
Comment (dmezh) [inline startup.sh:2]: Maybe leave a comment explaining the script

**Review by dmezh — CHANGES_REQUESTED** [review]

**Comment by dmezh** [comment]
Merging this will be huge! Exciting work 😄

**Comment by dmezh** [comment]
What's the status on this?

**Review by dmezh — CHANGES_REQUESTED** [review]
Please address the outstanding comments.

**Review by dmezh — APPROVED** [review]
Approving on condition that outstanding comments get resolved later over time.

**Pull request merged**
